utils/config/tokenrefresh.go

```diff
@@ -113,16 +143,36 @@ func tokenRefreshHandler(currentAccessToken string) (newAccessToken string, err
 		log.Debug("Token refreshed successfully.")
 	}
 
-	err = writeNewTokens(serverConfiguration, tokenRefreshServerId, newToken.AccessToken, newToken.RefreshToken)
+	err = writeNewArtifactoryTokens(serverConfiguration, tokenRefreshServerId, newToken.AccessToken, newToken.RefreshToken)
+	if err != nil {
+		log.Error("Failed writing new tokens to config after handling access token expiry: " + err.Error())
+	}
+	return newToken.AccessToken, nil
+}
+
+func refreshAccessTokenAndWriteToConfig(serverConfiguration *ServerDetails, currentAccessToken string) (string, error) {
+	refreshToken := serverConfiguration.RefreshToken
+	// Remove previous tokens
+	serverConfiguration.AccessToken = ""
+	serverConfiguration.RefreshToken = ""
+	// Try refreshing tokens
+	newToken, err := refreshExpiredAccessToken(serverConfiguration, currentAccessToken, refreshToken)
+
+	if err != nil {
+		log.Debug("Refresh access token failed: " + err.Error())
```

Comment:
There's probably no need for thid debug log message. Instead, let's add this message to the error that is returned here.